ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+Menu.swift

```diff
@@ -640,16 +648,23 @@ extension BrowserViewController {
         actionCopy.state = Preferences.General.nightModeEnabled.value
         return .updateAction(actionCopy)
       },
-      .init(id: .shredData) { @MainActor [unowned self] _ in
-        self.dismiss(animated: true) {
-          guard let tab = self.tabManager.selectedTab, let url = tab.url else { return }
-          let alert = UIAlertController.shredDataAlert(url: url) { _ in
-            self.shredData(for: url, in: tab)
+    ]
+    if BraveCore.FeatureList.kBraveShredFeature.enabled, let tabURL = tabManager.selectedTab?.url {
+      actions.append(
```

Comment:
Need to keep this action available even with a nil selected url, better to only feature flag check it then inside if-scope do

```swift
let isShredAvailable = tabManager.selectedTab?.url?.isShredAvailable ?? false
actions.append(...)
```

Reply:
Updated in [fc6eaa6](https://github.com/brave/brave-core/pull/27422/commits/fc6eaa66ca1f9f8b5e648415d28836183f97c932)